File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/fragments/investigation/utilities/codex/CodexFragment.java
package com.TritiumGaming.phasmophobiaevidencepicker.activities.fragments.investigation.utilities.codex;

import android.annotation.SuppressLint;
import android.content.res.Resources;
import android.os.Bundle;
import android.text.Html;
import android.util.TypedValue;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.GridLayout;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatTextView;
import androidx.navigation.Navigation;

import com.TritiumGaming.phasmophobiaevidencepicker.R;
import com.TritiumGaming.phasmophobiaevidencepicker.activities.fragments.investigation.InvestigationFragment;
import com.google.android.material.card.MaterialCardView;

public class CodexFragment extends InvestigationFragment {

    @Nullable
    @Override
    public View onCreateView(
            @NonNull LayoutInflater inflater,
            @Nullable ViewGroup container,
            @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_utilities_codex, container, false);
    }

    @SuppressLint("UseCompatLoadingForDrawables")
    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {

        super.onViewCreated(view, savedInstanceState);

        GridLayout gridView = view.findViewById(R.id.grid_codex);

        MaterialCardView gotoCodex = gridView.findViewById(R.id.grid_codexmenu_option1);
        gotoCodex.setOnClickListener(v -> Navigation.findNavController(v)
                .navigate(R.id.action_codexFragment_to_equipmentStoreFragment));

        MaterialCardView gotoCursedPossessions = gridView.findViewById(R.id.grid_codexmenu_option2);
        gotoCursedPossessions.setOnClickListener(v -> Navigation.findNavController(v)
                .navigate(R.id.action_codexFragment_to_cursedPossessionsFragment));

        TypedValue typedValue = new TypedValue();
        Resources.Theme theme = getContext().getTheme();
        theme.resolveAttribute(R.attr.codex_2, typedValue, true);
        int color1 = typedValue.data;
        theme.resolveAttribute(R.attr.codex_4, typedValue, true);
        int color2 = typedValue.data;
        String color1Hex = String.format("#%06X", (0xFFFFFF & color1));
        String color2Hex = String.format("#%06X", (0xFFFFFF & color2));

        AppCompatTextView label_ghostOS = view.findViewById(R.id.label_codex_ghostos);
        label_ghostOS.setText(Html.fromHtml(getString(R.string.codex_label_gh_ost)
                .replaceAll("#99AEB3", color1Hex)
                .replaceAll("#FFB43D", color2Hex)));
    }

    @Override
    public void softReset() {

    }

}

File: app-compose/src/main/java/com/tritiumgaming/phasmophobiaevidencepicker/ui/noncompose/investigation/utilities/codex/CodexFragment.kt
package com.tritiumgaming.phasmophobiaevidencepicker.ui.noncompose.investigation.utilities.codex

import android.text.Html
import androidx.appcompat.widget.AppCompatTextView
import com.tritiumgaming.phasmophobiaevidencepicker.R
import com.tritiumgaming.phasmophobiaevidencepicker.ui.investigation.InvestigationFragment
import com.tritiumgaming.phasmophobiaevidencepicker.utils.ColorUtils.getColorFromAttribute

abstract class CodexFragment : InvestigationFragment() {
    fun stylizeLogo(ghostOSLabel: AppCompatTextView?) {

        val color1 = getColorFromAttribute(requireContext(), R.attr.codex3_gh0stTextNormal)
        val color2 = getColorFromAttribute(requireContext(), R.attr.codex4_gh0stTextAlt)
        val color1Hex = String.format("#%06X", (0xFFFFFF and color1))
        val color2Hex = String.format("#%06X", (0xFFFFFF and color2))

        ghostOSLabel?.text = Html.fromHtml(
            getString(R.string.codex_label_gh_ost)
                .replace("#99AEB3".toRegex(), color1Hex)
                .replace("#FFB43D".toRegex(), color2Hex)
        )
    }
}
